### PR TITLE
Introduce temporary voting process to approve jj governance structures

### PR DESCRIPTION
**Pull request by nasamuffin**
Hi! I'm Emily, part of the governance working group. I am not a regular contributor to jj, but part of my role at Google is to guide open source projects in Martin's and my organization, helping make sure our upstream communities are healthy and productive.

The governance working group (me, @arxanas, @thoughtpolice, and @martinvonz) wants to be able to get community approval on governance structure changes. Bootstrapping is hard - how do you get community approval for a process to get community approval? - so we're introducing this process by the normal pull request workflow.

Things to keep in mind:
* This process is temporary! It will go away, replaced by something longer-lasting and more robust.
* Until this PR is merged, the governance working group is open to suggestions to clarify the language. But again, this is a temporary process - it doesn't need to be perfect, it just needs to be good enough for a handful of uses and then it can be discarded.

Things I'm especially interested in during review:
* Is the wording clear and unambiguous? Is it easy to understand how this process is supposed to work?
* Did I do the documentation change correctly? Do we actually want this `governance/` dir? (FWIW, I ran `cargo doc` and it passed with warnings unrelated to this file, but I'll openly admit that this is the first time I've built anything in jj and I might have missed :) )

### CODE REVIEW

**Review by steveklabnik — APPROVED** [review]
This sounds great to me!

**Comment by PhilipMetzger** [comment]
While this is LGTM, shouldn't the temporary governance be RFC 0? So I'd move it under `docs/governance/rfcs/rfc-0-temporary-voting.md`.

**Comment by nasamuffin** [comment]
> While this is LGTM, shouldn't the temporary governance be RFC 0? So I'd move it under `docs/governance/rfc/rfc-0-temporary-voting.md`.

Oh interesting! Would you rather see this doc explicitly outline that new proposals should go in `docs/governance/rfc/` and approved proposals move into `docs/governance/`?

**Comment by PhilipMetzger** [comment]
> Oh interesting! Would you rather see this doc explicitly outline that new proposals should go in `docs/governance/rfc/` and approved proposals move into `docs/governance/`?

Since it lays it out as that we start a RFC process, kinda (it also reminded me of my comment here https://github.com/martinvonz/jj/pull/1411#issuecomment-1481578377). I'm not sure that they should move though, as each RFC could also have a "approved:" field with the final decision.

**Comment by nasamuffin** [comment]
> Since it lays it out as that we start a RFC process

Kinda. But kinda not :) It's a temporary process - one of the things we want to use this process to approve is an actual formal RFC process. So I don't know if I want to unilaterally guess what that RFC process is (that's one thing the working group hasn't talked about yet pretty much at all). Although it's easy to move things around later if we change our minds.

**Comment by gulbanana** [comment]
looks pretty good. certainly adequate for a bootstrap process

**Review by eopb — APPROVED** [review]

**Comment by ilyagr** [comment]
> Did I do the documentation change correctly? Do we actually want this governance/ dir? (FWIW, I ran cargo doc and it passed with warnings unrelated to this file, but I'll openly admit that this is the first time I've built anything in jj and I might have missed :) )

Since CI passed, we know that this builds fine, but if you really wanted to check it manually, you'd need to use `mkdocs build` or `mkdocs serve` as explained in the [contributing doc](https://martinvonz.github.io/jj/prerelease/contributing/#previewing-the-html-documentation). I think it's not really necessary; the main reason to do it would be to check that it will look good on the website, but it's very likely fine, we can make cosmetic changes later, and most importantly it's unlikely to confuse users even if something is wrong.

File: docs/governance/temporary-voting.md
Comment (ilyagr) [inline docs/governance/temporary-voting.md:128]: Minor note: when possible and for larger proposals, it might be nice to have the option of the voting lasting longer than a week, e.g. to include 2 weekends, or even closer to 2 weeks.

So, I'd adjust these rules to something like:

- Voting is open for a minimum of 1 week, longer when appropriate (for less urgent, larger and/or more complex proposals).
- The deadline is announced at the beginning of the voting period.

That said, I'm OK with the proposal as is if anyone prefers that.

**Review by ilyagr — APPROVED** [review]
Looks great to me.

I'm approving this in the "if it were up to me..." sense, not in the "I'm sure everyone agrees" sense.

File: docs/governance/temporary-voting.md
Comment (nasamuffin) [reply docs/governance/temporary-voting.md:128]: Yeah, that seems fine to me. I'm going to rephrase it to put an explicit mention of capturing two weekends as a reason to extend the voting deadline, will send a new version sometime today.

**Pull request merged**
